### PR TITLE
Crankydillo 661 perpstream matvalues

### PR DESCRIPTION
**Pull request by akara**
Sorry some of the other branches leaked in and I don't have the time to clean up. So please do not totally merge. Please just take the changes on your file. I've tested them.

### CODE REVIEW

File: squbs-unicomplex/src/test/scala/org/squbs/stream/PerpetualStreamMatValueSpec.scala [hunk 1]
```diff
@@ -127,18 +132,18 @@ object PerpetualStreamMatValueSpecHelper {
...
-  def useSystem[PC <: PerpStream[_]](perpStream: Class[PC])(fn: Try[ActorRef] => Unit): Unit = {
-    implicit val actorSystem = ActorSystem()
-    implicit val materializer = ActorMaterializer()
+  def useSystem[PC <: PerpStream[_] : ClassTag](fn: Try[ActorRef] => Unit)
+                                               (implicit system: ActorSystem, mat: Materializer): Unit = {
 
-    val perpRef = actorSystem.actorOf(Props(perpStream))
-    val someRef = actorSystem.actorOf(Props(new SomeActor(perpRef)))
+    val perpRef = system.actorOf(Props[PC])
+    val someRef = system.actorOf(Props(new SomeActor(perpRef)))
```
Comment (akara) [inline squbs-unicomplex/src/test/scala/org/squbs/stream/PerpetualStreamMatValueSpec.scala:139]: I'm reverting this line to what is was before. After double-checking the `Props` code, this is the correct way to create this actor. And no, it does not create the actor at the creation of props (as was the issue earlier). It takes this `by-name` parameter and executes it when the actor is indeed created. Type inference allows us not to have to pass a type parameter here. And the `ClassTag` is generated by the compiler to fill in the type information. In full the signature would look like `Props[SomeActor] { new SomeActor(perpRef) }`. Ingenious!

**Comment by crankydillo** [comment]
@akara My apologies, but I'm just now noticing this.  Do I need to do anything with this or did we get all the changes we need in [this PR](https://github.com/paypal/squbs/pull/692)?

**Comment by akara** [comment]
I don't recall any more. Lets discuss tomorrow.

**Comment by crankydillo** [comment]
@akara Ah, you [note](https://github.com/crankydillo/squbs/pull/1#issue-243464991) that I should just take the relevant changes, which I did ([example](https://github.com/crankydillo/squbs/blob/77e2589050a3aad99e66b2f5bba225e9fcab9953/squbs-unicomplex/src/test/scala/org/squbs/stream/PerpetualStreamMatValueSpec.scala#L67)).  Just reopen if you disagree.

**Pull request closed without merging**